### PR TITLE
build.d: Allow passing additional DFLAGS,CXXFLAGS as arguments

### PR DESCRIPTION
**Pull request by MoonlightSentinel**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
```
build.d(1239): Error: mutable method `std.algorithm.searching.findSplit!("a == b", string, string).findSplit.Result!(string, string).Result.opCast!bool.opCast` is not callable using a `const` object
```

File: src/build.d [hunk 1]
```diff
@@ -1228,14 +1234,21 @@ void args2Environment(ref string[] args)
...
-        auto key = sp.front;
-        auto value = sp.dropOne.front;
-        environment[key] = value;
-        env[key] = value;
+        const key = parts[0];
+        const value = parts[2];
+
+        if (key.among("DFLAGS", "CXXFLAGS")) {
+            flags[key] = value.split();
+        }
```
Comment (marler8997) [inline src/build.d]: I haven't seen this kind of `if else` structure before, can you normalize it?

File: src/build.d [hunk 2]
```diff
@@ -898,7 +897,11 @@ void processEnvironment()
     {
         dflags ~= ["-fsanitize="~env["ENABLE_SANITIZERS"]];
     }
-    flags["DFLAGS"] ~= dflags;
+
+    if (auto ptr = "DFLAGS" in flags) // Retain user-defined flags
```
Comment (marler8997) [inline src/build.d]: What about:
```D
flags["DFLAGS"] = flags.get("DFLAGS", "") ~ dflags;
```

**Comment by marler8997** [comment]
Very nice, I can add flags to each compiler invocation with:
```
./build.d "DFLAGS=-v -whateverelse ..."
```

File: src/build.d (hunk 1 above)
Comment (MoonlightSentinel) [reply src/build.d]: Thats an old habit, changed to phobos style

File: src/build.d (hunk 2 above)
Comment (MoonlightSentinel) [reply src/build.d]: I initially wanted to use `update`  but hit an ldc bug. Changed to your approach.

**Comment by MoonlightSentinel** [comment]
> ```
> build.d(1239): Error: mutable method `std.algorithm.searching.findSplit!("a == b", string, string).findSplit.Result!(string, string).Result.opCast!bool.opCast` is not callable using a `const` object
> ```

Arrgh, that bug was present until 2.084 ... changed `const` to `auto`

**Review by marler8997 — APPROVED** [review]

**Comment by MoonlightSentinel** [comment]
Updated the DDOC comment to reflect the current behaviour

**Comment by MoonlightSentinel** [comment]
Ping @thewilsonator

**Pull request merged**